packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -252,11 +254,57 @@ class AndroidGradleBuilder implements AndroidBuilder {
     );
   }
 
-  Future<RunResult> _runGradleTask(
+  Future<int> _runGradleTask(
     String taskName, {
     List<String> options = const <String>[],
-    required FlutterProject project
+    required FlutterProject project,
+    required List<GradleHandledError> localGradleErrors,
+    int retry = 0,
+    VoidCallback? preRunTask,
+    VoidCallback? postRunTask,
+    int? maxRetries,
```

Comment:
I was talking to @johnmccutchan and he indicated we should be removing retries from our system not adding in layers of retries.  I think there is merit to the rest of this pr without the retry logic. 

Are we hitting the network or something else here that would cause us to expect flakey connnections.

Reply:
The gradle error handlers are defined in https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/android/gradle_errors.dart. Retries are triggered for the handlers that return `GradleBuildStatus.retry`. It looks like there might be a couple that are not network related, but the file seems to have comments with context that would help a deeper audit.

Reply:
> zipExceptionHandler <- corrupted local cache
I suspect this is due to a failed download (we unzip right after downloading)